### PR TITLE
DOCS-938: Document Vision Service API

### PR DESCRIPTION
**Pull request by JessamyT**
Also fix some copy pasta errors and missing params

### CODE REVIEW

File: docs/services/vision/classification.md [hunk 1]
```diff
@@ -137,14 +137,14 @@ cam1 = Camera.from_robot(robot, "cam1")
 # Grab Viam's vision service for the classifier
 my_classifier = VisionClient.from_robot(robot, "my_classifier")
 
-img = await cam1.get_image()
-classifications = await my_classifier.get_classifications_from_camera(img)
+# Get the top 2 most confident classifications from the camera output
```
Comment (npentrel) [inline docs/services/vision/classification.md:140]: I don't think a classification can be confident. How about
```suggestion
# Get the top 2 classifications with the highest confidence scores from the camera output
```

**Review by npentrel — APPROVED** [review]
LGTM from docs side

File: docs/services/vision/classification.md [hunk 2]
```diff
@@ -174,8 +174,8 @@ if err != nil {
     logger.Fatalf("Cannot get Vision Service: %v", err)
 }
 
-// Get classifications from the camera output
-classifications, err := visService.ClassificationsFromCamera(context.Background(), myCam, nil)
+// Get the top 2 most confident classifications from the camera output
```
Comment (JessamyT) [inline docs/services/vision/classification.md:177]:
```suggestion
// Get the top 2 classifications with the highest confidence scores from the camera output
```

File: docs/services/vision/classification.md [hunk 3]
```diff
@@ -195,7 +195,8 @@ img, release, err := camStream.Next(context.Background())
 defer release()
 
 // Apply the color classifier to the image from your camera (configured as "cam1")
-classificationsFromImage, err := visService.GetClassifications(context.Background(), img, nil)
+// Get the top 2 most confident classifications
```
Comment (JessamyT) [inline docs/services/vision/classification.md:198]:
```suggestion
// Get the top 2 classifications with the highest confidence scores
```

File: docs/services/vision/_index.md [hunk 4]
```diff
@@ -14,3 +15,416 @@ While the camera component lets you access what your robot's camera sees, the Vi
...
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/services/vision/client/index.html#viam.services.vision.client.VisionClient.get_object_point_clouds).
+
+```python {class="line-numbers linkable-line-numbers" data-line="8"}
+# Grab the 3D camera from the robot
+cam1 = Camera.from_robot(robot, "cam1")
+
+# Grab the object segmenter you configured on your robot
+my_segmenter = VisionClient.from_robot(robot, "my_segmenter")
+
+# Get the top 2 classifications from the next image from the camera
```
Comment (JessamyT) [inline docs/services/vision/_index.md:385]:
```suggestion
# Get the 2 classifications with the highest confidence scores from the next image from the camera
```

File: docs/services/vision/_index.md [hunk 5]
```diff
@@ -14,3 +15,416 @@ While the camera component lets you access what your robot's camera sees, the Vi
...
+  logger.Fatalf("cannot get camera: %v", err)
+}
+
+// Grab the classifier you configured on your robot
+visService, err := vision.from_robot(robot=robot, name='my_classifier')
+if err != nil {
+    logger.Fatalf("Cannot get Vision Service: %v", err)
+}
+
+// Get the top 2 classifications from the camera output
```
Comment (JessamyT) [inline docs/services/vision/_index.md:347]:
```suggestion
// Get the 2 classifications with the highest confidence scores from the camera output
```

File: docs/services/vision/_index.md [hunk 6]
```diff
@@ -14,3 +15,416 @@ While the camera component lets you access what your robot's camera sees, the Vi
...
+}
+
+// Get the stream from a camera
+camStream, err := myCam.Stream(context.Background())
+
+// Get an image from the camera stream
+img, release, err := camStream.Next(context.Background())
+defer release()
+
+// Get the top 2 classifications from the image
```
Comment (JessamyT) [inline docs/services/vision/_index.md:271]:
```suggestion
// Get the 2 classifications with the highest confidence scores from the next image from the camera
```

File: docs/services/vision/_index.md [hunk 7]
```diff
@@ -14,3 +15,416 @@ While the camera component lets you access what your robot's camera sees, the Vi
...
+cam1 = Camera.from_robot(robot, "cam1")
+
+# Get the classifier you configured on your robot
+my_classifier = VisionClient.from_robot(robot, "my_classifier")
+
+# Get an image from the camera
+img = await cam1.get_image()
+
+# Get classifications from that image
+classifications = await my_classifier.get_classifications(img)
```
Comment (JessamyT) [inline docs/services/vision/_index.md:229]:
```suggestion
classifications = await my_classifier.get_classifications(img, 2)
```

**Review by bhaney — APPROVED** [review]
LGTM!

File: docs/services/vision/_index.md [hunk 8]
```diff
@@ -14,3 +15,416 @@ While the camera component lets you access what your robot's camera sees, the Vi
...
+{{% tab name="Python" %}}
+
+**Parameters:**
+
+- `image` [(RawImage)](https://python.viam.dev/autoapi/viam/media/video/index.html#viam.media.video.RawImage): The image in which to look for detections.
+- `extra` (Mapping[str, Any]) (*optional*): A generic struct, containing extra options to pass to the underlying RPC call.
+
+**Returns:**
+
+- [(List[Detection])](https://python.viam.dev/autoapi/viam/proto/service/vision/index.html#viam.proto.service.vision.Detectionn): A list of 2D bounding boxes, their labels, and the confidence score of the labels around the detected objects, and confidence scores of those detections.
```
Comment (sguequierre) [inline docs/services/vision/_index.md:58]:
```suggestion
- [(List[Detection])](https://python.viam.dev/autoapi/viam/proto/service/vision/index.html#viam.proto.service.vision.Detection): A list of 2D bounding boxes, their labels, and the confidence score of the labels around the detected objects, and confidence scores of those detections.
```

**Review by sguequierre — COMMENTED** [review]
This mostly LGTM from docs side & will be a huge improvement to this page. Just chiming in to suggest a link fix and also suggest if you have time-- it might be a comprehensive improvement on these docs to keep the API methods for the different "types" of vision service separate on those three pages? 

Or just some way more clearly delineating the vision service configuration options?
The difference between `detector`, `classifier`, `segmenter` API usage is somewhat confusing when organized together to me. I'll make a note of this for the docs sync so we can discuss this briefly then if you want!

**Comment by JessamyT** [comment]
Fair point @sguequierre! Since the API is on a single page for other components and services, I think it makes sense to leave it on one page for this too, and take a tip from Movement Sensor. I've just made a commit to add a column to the API overview table, specifying which models each method is supported by. And, I've added links to the different models on each method description. Hopefully that improves clarity without burying the API too much. If you want to discuss splitting the API into different model pages, yeah let's discuss at the meeting and that can happen in a different PR if that's what we decide on!?

File: docs/services/vision/_index.md [hunk 9]
```diff
@@ -14,3 +15,416 @@ While the camera component lets you access what your robot's camera sees, the Vi
...
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `cameraName` [(string)](https://pkg.go.dev/builtin#string): The name of the camera from which to get an image to run the classifier on.
+- `n` [(int)](https://pkg.go.dev/builtin#int): The number of classifications to return.
+For example, if you specify `3` you will get the top three classifications with the greatest confidence scores.
+- `extra` [(map\[string\]interface{})](https://go.dev/blog/maps): Extra options to pass to the underlying RPC call.
+
+**Returns:**
+
+- [([]objection.Classification)](https://pkg.go.dev/go.viam.com/vision/classification#Classifications): A list of classification labels, and the confidence scores of those classifications.
```
Comment (npentrel) [inline docs/services/vision/_index.md:328]:
```suggestion
- [([]objection.Classification)](https://pkg.go.dev/go.viam.com/rdk/vision/classification#Classifications): A list of classification labels, and the confidence scores of those classifications.
```

File: docs/services/vision/_index.md [hunk 10]
```diff
@@ -14,3 +15,416 @@ While the camera component lets you access what your robot's camera sees, the Vi
...
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `cameraName` [(string)](https://pkg.go.dev/builtin#string): The name of the 3D camera from which to get point cloud data.
+- `extra` [(map\[string\]interface{})](https://go.dev/blog/maps): Extra options to pass to the underlying RPC call.
+
+**Returns:**
+
+- [([]objection.Classification)](https://pkg.go.dev/go.viam.com/vision#Object): A list of point clouds and associated metadata like the center coordinates of each point cloud.
```
Comment (npentrel) [inline docs/services/vision/_index.md:400]:
```suggestion
- [([]objection.Classification)](https://pkg.go.dev/go.viam.com/rdk/vision#Object): A list of point clouds and associated metadata like the center coordinates of each point cloud.
```

**Pull request merged**